src/main/java/de/konfidas/ttc/validation/TransactionCounterValidator.java

```diff
@@ -0,0 +1,120 @@
+package de.konfidas.ttc.validation;
+
+import de.konfidas.ttc.exceptions.LogMessageValidationException;
+import de.konfidas.ttc.exceptions.ValidationException;
+import de.konfidas.ttc.messages.LogMessageImplementation;
+import de.konfidas.ttc.messages.TransactionLogMessage;
+import de.konfidas.ttc.tars.LogMessageArchive;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+
+// FIXME: this is incomplete and does not work fully.
+public class TransactionCounterValidator implements Validator {
+    final HashMap<BigInteger,OpenTransaction> openTransactions;
+    BigInteger transactionCounter;
+
+    TransactionCounterValidator(){
+        openTransactions = new HashMap<>();
+        transactionCounter = BigInteger.ONE;
+    }
+
+    public Collection<ValidationException> validate(LogMessageArchive tar){

```

Comment:
Auch hier komme ich bei der Prüfung nicht ganz mit => TElefon

Reply:
The TransactionCounterValidator is still work in progress and incomplete. So we can discuss this as soon as it is done fully ;)